### PR TITLE
Publish server stream when fee changes (RIPD-1406)

### LINKED ISSUE #1991: server streaming not working.
was trying to subscribe to server status stream via websocket command as follow:
[https://ripple.com/build/websocket-tool/#subscribe](https://ripple.com/build/websocket-tool/#subscribe)

but the subscription never received any stream of server status.

### PR DESCRIPTION
**Pull request by bachase**
Resolves #1991

Publish a server status update after every ledger close if there is a change in fees.

### CODE REVIEW

File: src/ripple/app/misc/NetworkOPs.cpp [hunk 1]
```diff
@@ -1600,14 +1648,45 @@ void NetworkOPsImp::pubServer ()
...
         Json::Value jvObj (Json::objectValue);
-        auto const& feeTrack = app_.getFeeTrack();
 
-        jvObj [jss::type]          = "serverStatus";
+        ServerFeeSummary f{app_.openLedger().current()->fees().base,
+            app_.getTxQ().getMetrics(*app_.openLedger().current()),
+            app_.getFeeTrack()};
+
+        // Need to cap to uint64 to uint32 due to JSON limitations
+        auto clamp = [](std::uint64_t v)
```
Comment (mellery451) [inline src/ripple/app/misc/NetworkOPs.cpp]: can you use the impl in beast for this?   https://github.com/vinniefalco/Beast/blob/master/include/beast/core/detail/clamp.hpp

File: src/ripple/app/misc/NetworkOPs.cpp [hunk 2]
```diff
@@ -1600,14 +1648,45 @@ void NetworkOPsImp::pubServer ()
...
+        {
+            constexpr std::uint64_t max32 =
+                std::numeric_limits<std::uint32_t>::max();
+
+            return static_cast<std::uint32_t>(std::min(max32, v));
+        };
+
+
+        jvObj [jss::type] = "serverStatus";
```
Comment (mellery451) [inline src/ripple/app/misc/NetworkOPs.cpp]: I have no idea if we have standardized this, but it seems to me like the strings we return in JSON are typically all lower or are upper camel case (e.g "ServerStatus"). Just something to consider.

File: src/ripple/app/misc/NetworkOPs.cpp (hunk 1 above)
Comment (bachase) [inline src/ripple/app/misc/NetworkOPs.cpp]: Yes, nice pointer!

File: src/ripple/app/misc/NetworkOPs.cpp (hunk 2 above)
Comment (bachase) [inline src/ripple/app/misc/NetworkOPs.cpp]: The lower camel case seems to be the convention for subscriptions, see https://ripple.com/build/rippled-apis/#subscribe

Comment (mellery451) [inline src/ripple/app/misc/NetworkOPs.cpp]: 👍

File: src/ripple/app/misc/NetworkOPs.cpp (hunk 1 above)
Comment (bachase) [inline src/ripple/app/misc/NetworkOPs.cpp]: Actually, JSON needs to clamp down to std::uint32_t, but that clamp is using std::size_t which is bigger on many platforms.

**Comment by bachase** [comment]
Thanks for your feedback @ximinez.  I agree that the additional reporting spot better addresses the real time desire for fee changes.  In a short test run in `rippled`, it published 91 updates in 3 minutes, versus just 7 updates when only reporting post ledger close.

**Comment by bachase** [comment]
Rebased on 0.60.0-b5

**Comment by seelabs** [comment]
In 0.60.0-b7

**Pull request closed without merging**